File: javascript/phone-number/phone-number.js
var PhoneNumber = function(input){
  this.input = input;
  this.digits = this.number();
};

PhoneNumber.prototype.number = function(){
  var digits = this.input.replace(/\D/g,'');
  if (digits.length === 11 && digits.charAt(0) === '1') { return digits.substring(1) };
  if (digits.length > 10 || digits.length < 10) { return '0000000000' };
  return digits;
};

PhoneNumber.prototype.areaCode = function(){
  return this.digits.substring(0, 3);
};

PhoneNumber.prototype.toString = function(){
  var formattedAreaCode = '(' + this.areaCode() + ') '
  var formattedNumber = this.digits.substring(3, 6) + '-' + this.digits.substring(6, 11);
  return formattedAreaCode + formattedNumber;
};

module.exports = PhoneNumber;

File: java/phone-number/src/main/java/PhoneNumber.java
public class PhoneNumber {

    String rawDigits;

    public PhoneNumber(String rawDigits) {
        this.rawDigits = rawDigits;
    }

    public String getNumber() {
        String digits = rawDigits.replaceAll("\\D", "");
        if (digits.length() == 10) { return digits; }
        if (digits.length() == 11 && digits.charAt(0) == '1') { return digits.substring(1); }
        return "0000000000";
    }

    public String getAreaCode() {
        return getNumber().substring(0, 3);
    }

    public String pretty() {
        String digits = getNumber();
        return "(" + digits.substring(0,3) + ") " + digits.substring(3,6) + "-" + digits.substring(6,10);

    }
}